### PR TITLE
docs(guides) updates to code splitting guide

### PR DESCRIPTION
**Pull request by EugeneHlushko**
* Remove duplicated warning
* Rename `SplitChunksPlugin` mentions to correspond to its name and not to configuration option.
* Various linking improvements
* Remove outdated `CommonChunksPlugin` reference

### CODE REVIEW

File: src/content/guides/code-splitting.md [hunk 1]
```diff
@@ -147,7 +145,7 @@ __webpack.config.js__
   };
 ```
 
-With the [`SplitChunks`](/plugins/split-chunks-plugin/) in place, we should now see the duplicate dependency removed from our `index.bundle.js` and `another.bundle.js`. The plugin should notice that we've separated `lodash` out to a separate chunk and remove the dead weight from our main bundle. Let's do an `npm run build` to see if it worked:
+With the [`optimization.SplitChunks`](/configuration/optimization/#optimization-splitchunks) configuration option in place, we should now see the duplicate dependency removed from our `index.bundle.js` and `another.bundle.js`. The plugin should notice that we've separated `lodash` out to a separate chunk and remove the dead weight from our main bundle. Let's do an `npm run build` to see if it worked:
```
Comment (montogeek) [inline src/content/guides/code-splitting.md]: Why add `optimization`?

Comment (EugeneHlushko) [reply src/content/guides/code-splitting.md]: by the text, `With X in place, we should see something removed...`, we dont put SplitChunksPlugin in place, we enable it via optimization configuration options...

Comment (EugeneHlushko) [reply src/content/guides/code-splitting.md]: So i thought it is clearer to actually show users the configuration option. But your point is very correct in regard of wrong URL, it should be: `/plugins/split-chunks-plugin/#optimization-splitchunks` and not the `optimization` page

Comment (EugeneHlushko) [reply src/content/guides/code-splitting.md]: there is also a conflict now, need to see what changed and resolve

File: src/content/guides/code-splitting.md [hunk 2]
```diff
@@ -145,7 +145,7 @@ __webpack.config.js__
   };
 ```
 
-With the [`SplitChunks`](/plugins/split-chunks-plugin/) in place, we should now see the duplicate dependency removed from our `index.bundle.js` and `another.bundle.js`. The plugin should notice that we've separated `lodash` out to a separate chunk and remove the dead weight from our main bundle. Let's do an `npm run build` to see if it worked:
+With the [`optimization.splitChunks`](/plugins/split-chunks-plugin/#optimization-splitchunks) configuration option in place, we should now see the duplicate dependency removed from our `index.bundle.js` and `another.bundle.js`. The plugin should notice that we've separated `lodash` out to a separate chunk and remove the dead weight from our main bundle. Let's do an `npm run build` to see if it worked:
```
Comment (EugeneHlushko) [inline src/content/guides/code-splitting.md]: @montogeek updated, also resolved a conflict

File: src/content/guides/code-splitting.md [hunk 3]
```diff
@@ -113,14 +113,14 @@ As mentioned there are some pitfalls to this approach:
...
+The first of these two points is definitely an issue for our example, as `lodash` is also imported within `./src/index.js` and will thus be duplicated in both bundles. Let's remove this duplication by using the [`SplitChunksPlugin`](/plugins/split-chunks-plugin/).
 
 
 ## Prevent Duplication
 
-W> The CommonsChunkPlugin has been removed in webpack v4 legato. To learn how chunks are treated in the latest version, check out the [SplitChunksPlugin](/plugins/split-chunks-plugin/).
+The [`SplitChunksPlugin`](/plugins/split-chunks-plugin/) allows us to extract common dependencies into an existing entry chunk or an entirely new chunk. Let's use this to de-duplicate the `lodash` dependency from the previous example:
 
-The [`SplitChunks`](/plugins/split-chunks-plugin/) allows us to extract common dependencies into an existing entry chunk or an entirely new chunk. Let's use this to de-duplicate the `lodash` dependency from the previous example:
+W> The `CommonsChunkPlugin` has been removed in webpack v4 legato. To learn how chunks are treated in the latest version, check out the [`SplitChunksPlugin`](/plugins/split-chunks-plugin/).
```
Comment (EugeneHlushko) [inline src/content/guides/code-splitting.md]: i've moved this further from header (it was right after `Prevent duplication`) to be more reader-friendly

**Comment by montogeek** [comment]
Thanks!

**Pull request merged**